ddtrace/contrib/rq/patch.py

```diff
@@ -0,0 +1,201 @@
+import os
+
+import rq
+
+from ddtrace import Pin
+from ddtrace import config
+from ddtrace.constants import SPAN_KIND
+from ddtrace.internal.constants import COMPONENT
+from ddtrace.internal.schema import schematize_messaging_operation
+from ddtrace.internal.schema import schematize_service_name
+from ddtrace.internal.schema.span_attribute_schema import SpanDirection
+from ddtrace.internal.utils.version import parse_version
+
+from .. import trace_utils
+from ...ext import SpanKind
+from ...ext import SpanTypes
+from ...internal.utils import get_argument_value
+from ...internal.utils.formats import asbool
+from ...propagation.http import HTTPPropagator
+
+
+config._add(
+    "rq",
+    dict(
+        distributed_tracing_enabled=asbool(os.environ.get("DD_RQ_DISTRIBUTED_TRACING_ENABLED", True)),
+        _default_service=schematize_service_name("rq"),
+    ),
+)
+
+config._add(
+    "rq_worker",
+    dict(
+        distributed_tracing_enabled=asbool(os.environ.get("DD_RQ_DISTRIBUTED_TRACING_ENABLED", True)),
+        _default_service=schematize_service_name("rq-worker"),
+    ),
+)
+
+rq_version_str = getattr(rq, "__version__", "0.0.0")
+rq_version = parse_version(rq_version_str)
```

Comment:
Instead of moving everything out of `__init__.py` the alternative could have been to just put this in a function.

Since most of the methods in here accept `rq` as an argument rather than relying on the globally imported module.

```python
def get_rq_version(rq):
    rq_version_str = getattr(rq, "__version__", "0.0.0")
    return rq_version = parse_version(rq_version_str)

... 


rq_version = get_rq_version(rq)
if rq_version >= (1, 14, 0):
    pass
else:
    pass

```


And since we only ever use the rq version from `patch`/`unpatch` we don't have to worry too much about the overhead of `get_rq_version()` calls.

What you have here also works, but if you want an approach that is less changes, this is an approach we could take.

Reply:
I thought I'd align on the "usual" way to structure the contrib, in the hopes that this would make refactors easier, especially given @emmettbutler 's current work on contribs.